scripts/db-invariant-sweeps/predicates/section9-po-ap-controls.sql

```
@@ -48,14 +75,23 @@ UNION ALL
-   OR has_table_privilege('anon', 'public.vendors', 'DELETE')
-   OR has_table_privilege('anon', 'public.vendors', 'TRUNCATE')
+-- has_table_privilege() with a role NAME raises `role "<name>" does not exist`
+-- when that role is absent. The error aborts the whole statement, so every other
+-- arm of this UNION ALL returns nothing and the sweep reports a false CLEAN.
+--
+-- An `EXISTS (...) AND has_table_privilege('name', ...)` guard is NOT enough: SQL
+-- does not promise left-to-right AND evaluation, so the planner may still evaluate
+-- the privilege call and raise. Pass the role's OID from a scalar subquery instead.
+-- has_table_privilege is strict, so an absent role yields NULL — never TRUE, never
+-- an error — which is the correct outcome: a role that does not exist holds nothing.
+WHERE has_table_privilege((SELECT r.oid FROM pg_roles r WHERE r.rolname = 'authenticated'), 'public.vendors', 'INSERT')
```

SQLFluff (4.3.0)

[error] The 'WHERE' keyword should always start a new line.

(LT14)